src/batched/KokkosBatched_Trsm_Serial_Impl.hpp

```diff
@@ -297,6 +297,196 @@ namespace KokkosBatched {
       }
     };
 
+    ///
+    /// L/L/T
+    ///
+    /// B := inv(tril(AT)) (alpha*B)
+    /// A(m x m), B(m x n)
+
+#if                                                     \
+  defined(__KOKKOSBATCHED_INTEL_MKL__) &&               \
+  defined(__KOKKOSBATCHED_INTEL_MKL_BATCHED__) &&       \
+  defined(__KOKKOSBATCHED_INTEL_MKL_COMPACT_BATCHED__)    
+    template<typename ArgDiag>
+    struct SerialTrsm<Side::Left,Uplo::Lower,Trans::Transpose,ArgDiag,Algo::Trsm::CompactMKL> {
+      template<typename ScalarType,
+               typename AViewType,
+               typename BViewType>
+      KOKKOS_INLINE_FUNCTION
+      static int
+      invoke(const ScalarType alpha,
+             const AViewType &A,
+             const BViewType &B) {
+        typedef typename BViewType::value_type vector_type;
+        //typedef typename vector_type::value_type value_type;
+        
+        const int
+          m = B.dimension(0),
+          n = B.dimension(1);
+
+        static_assert(is_vector<vector_type>::value, "value type is not vector type");      
+        static_assert(vector_type::vector_length == 4 || vector_type::vector_length == 8, 
+                      "AVX, AVX2 and AVX512 is supported");
+        const MKL_COMPACT_PACK format = vector_type::vector_length == 8 ?  MKL_COMPACT_AVX512 : MKL_COMPACT_AVX;
+        
+        // no error check
+        int r_val = 0;
+        if (A.stride_0() == 1 && B.stride_0() == 1) {
+          mkl_dtrsm_compact(MKL_COL_MAJOR, 
+                            MKL_LEFT, MKL_LOWER, MKL_TRANS, 
+                            ArgDiag::use_unit_diag ? MKL_UNIT : MKL_NONUNIT,
+                            m, n, 
+                            alpha, 
+                            (const double*)A.data(), A.stride_1(), 
+                            (double*)B.data(), B.stride_1(), 
+                            format, (MKL_INT)vector_type::vector_length);
+        } else if (A.stride_1() == 1 && B.stride_1() == 1) {
+          mkl_dtrsm_compact(MKL_ROW_MAJOR, 
```

Comment:
I'll continue to express my concerns about exposing TPL details in header files.

Reply:
Thanks. Your connern is well acknowledged. mkl compact routines have the mkl prefix and using the mkl prefixed routines should not harm. However, I will encapsulate all other fortran blas and lapack interface which is highly possible to conflict with users declarations.

Reply:
@kyungjoo-kim I'm OK with that, as long as including the standard (non-batched) kokkos-kernels BLAS wrappers doesn't automaticallly include these headers.  Thanks!